ansible/tasks/initdb.yml

```diff
@@ -61,4 +92,17 @@
     password: "{{ db_password }}"
     force_basic_auth: yes
   with_items: "{{ db.authkeys }}"
-  when: result.status == 404
+  when: dbexists is defined and dbexists.status == 404
+
+- name: secure db against anonymous access
```

Comment:
but you are now conflating the initialization of the db with the configuration of the db provider. 
the flag you are setting applies only to couchdb, right? not subject db. also is there a reason to not use "secure" mode? even if you want this to be an additional task, it should be logically grouped within the couchdb role. after all, it does tweak the couch server.